### PR TITLE
Move to reusable workflows in GHA

### PR DESCRIPTION
**Pull request by viccuad**
Fixes: https://github.com/kubewarden/opa-policy-template/issues/1

### CODE REVIEW

**Review by jvanz — APPROVED** [review]

File: .github/workflows/test.yml.template [hunk 1]
```diff
@@ -0,0 +1,6 @@
+on: [push, pull_request]
+name: Continuous integration
+jobs:
+  test:
+    name: run tests and linters
+    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-rego.yml@v1
```
Comment (jvanz) [inline .github/workflows/test.yml.template:6]: I just noticed that in the [swift template](https://github.com/kubewarden/swift-policy-template/pull/2/files) you named the file as `test.yml`. Is there a reason why is different here?

Comment (viccuad) [reply .github/workflows/test.yml.template:6]: oh shoot. They should have been templates on the swift ones, too. I will open a PR for that. Thanks!

Comment (viccuad) [reply .github/workflows/test.yml.template:6]: Done in https://github.com/kubewarden/swift-policy-template/pull/3.

**Review by flavio — APPROVED** [review]

File: .github/workflows/test.yml.template (hunk 1 above)
Comment (jvanz) [reply .github/workflows/test.yml.template:6]: :)

**Pull request merged**